src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorVerifierInProcess.cs

```diff
@@ -235,16 +235,47 @@ public async Task CaretPositionAsync(int expectedCaretPosition, CancellationToke
             Assert.Equal(expectedCaretPosition, await TestServices.Editor.GetCaretPositionAsync(cancellationToken));
         }
 
-        public async Task ErrorTagsAsync(string[] expectedTags, CancellationToken cancellationToken)
+        public async Task ErrorTagsAsync(
+            (string errorType, TextSpan textSpan, string taggedText, string tooltipText)[] expectedTags, CancellationToken cancellationToken)
         {
             await TestServices.Workspace.WaitForAllAsyncOperationsAsync(
                 new[] { FeatureAttribute.Workspace, FeatureAttribute.SolutionCrawlerLegacy, FeatureAttribute.DiagnosticService, FeatureAttribute.ErrorSquiggles },
                 cancellationToken);
 
             var actualTags = await TestServices.Editor.GetErrorTagsAsync(cancellationToken);
-            AssertEx.EqualOrDiff(
-                string.Join(Environment.NewLine, expectedTags),
-                string.Join(Environment.NewLine, actualTags));
+            Assert.Equal(expectedTags.Length, actualTags.Length);
+            for (var i = 0; i < expectedTags.Length; i++)
```

Comment:
xUnit has some collection assert helpers which might be helpful here; it at least runs the assertions against all elements, so if each of them were to change you can see all the problems instead of just the first one. I imagine that was the reason the original EqualOrDiff was chosen, as this really does hurt debuggability. The other approach would be to keep this method signature as it is, and instead have a helper here that converts to strings and then compares the strings that way. But even the code as is a good improvement over having the tests themselves bake in the string format.